src/hooks/useHeaderSticky.tsx

```diff
@@ -0,0 +1,20 @@
+import { useState, useEffect, MutableRefObject } from 'react';
+
+export const useHeaderSticky = (containerRef: MutableRefObject<HTMLElement | null>) => {
+  const [isHeaderSticky, setIsHeaderSticky] = useState(false);
+  const containerEl = containerRef?.current;
+
+  const handleContainerScroll = () => {
```

Comment:
Memory leak issue: each time when `isHeaderSticky` is changed you will recieve new instance of `handleContainerScroll`. To avoid it you need to move this function within useEffect.